src/v/pandaproxy/server.cc

```diff
@@ -96,13 +96,40 @@ struct handler_adaptor : ss::httpd::handler_base {
                        [this, rq{std::move(rq)}, rp{std::move(rp)}]() mutable {
                            if (_ctx.as.abort_requested()) {
                                set_reply_unavailable(*rp.rep);
+                               _probe.increment_5xx();
                                return ss::make_ready_future<
                                  std::unique_ptr<ss::reply>>(std::move(rp.rep));
                            }
                            return _handler(std::move(rq), std::move(rp))
                              .then([](server::reply_t rp) {
                                  set_mime_type(*rp.rep, rp.mime_type);
                                  return std::move(rp.rep);
+                             })
+                             .handle_exception_type(
+                               [this](pandaproxy::parse::exception_base& ex) {
+                                   if (ex.error.value() >= 500) {
+                                       _probe.increment_5xx();
+                                   } else if (ex.error.value() >= 400) {
+                                       _probe.increment_4xx();
+                                   } else if (ex.error.value() >= 300) {
+                                       _probe.increment_3xx();
+                                   }
+
+                                   return errored_body(
+                                     ex.error, ss::sstring{ex.msg});
+                               })
+                             .handle_exception([this](std::exception_ptr eptr) {
+                                 ss::sstring msg;
+                                 try {
+                                     std::rethrow_exception(eptr);
+                                 } catch (const std::exception& ex) {
+                                     msg = ss::sstring{ex.what()};
+                                 }
+                                 vlog(plog.error, "Unknown error {}", msg);
+                                 _probe.increment_5xx();
+                                 return errored_body(
```

Comment:
Why not leverage `exception_reply`, which can be constructed directly from an `exception_ptr` and knows how to get the correct `status_code`, which can then be read here?

Reply:
To be honest, I didn't think about that. It even accounts for multiple error types! Good tip!

Reply:
this is no longer applicable since I changed the approach to use `exception_replier`